supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/DeclarationValidation.scala

```diff
@@ -0,0 +1,198 @@
+package cromwell.backend.impl.sfs.config
+
+import cromwell.backend.MemorySize
+import cromwell.backend.impl.sfs.config.ConfigConstants._
+import cromwell.backend.validation._
+import wdl4s.expression.NoFunctions
+import wdl4s.parser.MemoryUnit
+import wdl4s.types._
+import wdl4s.values.{WdlFloat, WdlInteger, WdlValue}
+import wdl4s.{Declaration, NoLookup, WdlExpression}
+
+/**
+  * Creates instances of runtime attribute validations from WDL declarations.
+  */
+object DeclarationValidation {
+  def fromDeclarations(declarations: Seq[Declaration]): Seq[DeclarationValidation] = {
+    declarations map fromDeclaration
+  }
+
+  /**
+    * Create a runtime attribute validation from a WDL declaration.
+    *
+    * @param declaration The declaration.
+    * @return The DeclarationValidation object for the declaration.
+    */
+  def fromDeclaration(declaration: Declaration): DeclarationValidation = {
+    declaration.name match {
+      // Docker and CPU are special keys understood by cromwell.
+      case DockerValidation.key => new DeclarationValidation(declaration, DockerValidation.instance)
+      case CpuValidation.key => new DeclarationValidation(declaration, CpuValidation.default)
+      // See MemoryDeclarationValidation for more info
+      case name if MemoryDeclarationValidation.isMemoryDeclaration(name) =>
+        new MemoryDeclarationValidation(declaration)
+      // All other declarations must be a Boolean, Float, Integer, or String.
```

Comment:
"Engine"-ish. `ContinueOnReturnCodeValidation` and `FailOnStderrValidation` are both required validations by the SFS.